### ISSUE TITLE
Integration without Home Assistant

### ISSUE THREAD

**Issue by rossnelson** [zigpy/bellows#259]
We are currently evaluating zigpy and bellows for use in our custom IoT solution. Our gateway runs containerized, so we are attempting to build a docker image that runs inside our k8s distribution alongside our zwave integration. So far, I haven't been able to get bellows to work with my Nortek HUSBZB-1. I get the same error whether installed at the system level or inside a docker image. So I'll include info on both and hopefully, someone will be able to point me in the right direction. 

### Docker

`Dockerfile`
```
FROM python:buster

RUN apt update \
  && apt install -y vim git build-essential \
  && pip install bellows \
  && mkdir -p /root/.confg/bellows \
  && touch /root/.confg/bellows/app.db

CMD ['bellows', 'form']
```

```
$ docker run \
  -it \
  --privileged=true \
  -v $(pwd)/data:/root/.config/bellows \
  -v /dev/ttyUSB1:/dev/ttyUSB1 \ # could vary per system/device
  -e EZSP_DEVICE=/dev/ttyUSB1 \
  {image id}
```

### System 

`$ cat /etc/os-release`
```
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
```

`$ pip3 show zigpy bellows`
```
Name: zigpy
Version: 0.20.1a1
Summary: Library implementing a ZigBee stack
Home-page: http://github.com/zigpy/zigpy
Author: Russell Cloran
Author-email: rcloran@gmail.com
License: GPL-3.0
Location: /home/pi/.local/lib/python3.7/site-packages
Requires: pycryptodome, voluptuous, aiohttp, crccheck
Required-by: bellows
---
Name: bellows
Version: 0.16.0
Summary: Library implementing EZSP
Home-page: http://github.com/zigpy/bellows
Author: Russell Cloran
Author-email: rcloran@gmail.com
License: GPL-3.0
Location: /home/pi/.local/lib/python3.7/site-packages
Requires: click-log, Click, pure-pcapy3, zigpy, pyserial-asyncio, voluptuous
Required-by:
```

The error I see when running `bellows form` in either the system or from docker is this:

```
Traceback (most recent call last):
  File "/usr/local/bin/bellows", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/bellows/cli/application.py", line 34, in form
    return util.app(inner, app_startup=False)(ctx)
  File "/usr/local/lib/python3.8/site-packages/bellows/cli/util.py", line 68, in inner
    loop.run_until_complete(async_inner(*args, **kwargs))
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.8/site-packages/bellows/cli/util.py", line 51, in async_inner
    application = await setup_application(
  File "/usr/local/lib/python3.8/site-packages/bellows/cli/util.py", line 136, in setup_application
    app = await bellows.zigbee.application.ControllerApplication.new(
  File "/usr/local/lib/python3.8/site-packages/zigpy/application.py", line 60, in new
    app = cls(config)
TypeError: Can't instantiate abstract class ControllerApplication with abstract methods probe
```

**Comment by Adminiuga** [comment]
There are some changes being made.  Use zigpy==0.20.1a3 and bellows 0.16.1a1
I think those should be published by now

**Comment by Adminiuga** [comment]
Fixed in `bellows==0.16.1`

`pip3 show zigpy bellows`
```
Name: zigpy
Version: 0.20.1
Summary: Library implementing a ZigBee stack
Home-page: http://github.com/zigpy/zigpy
Author: Russell Cloran
Author-email: rcloran@gmail.com
License: GPL-3.0
Location: /tmp/venv/lib/python3.7/site-packages
Requires: voluptuous, aiohttp, crccheck, pycryptodome
Required-by: bellows
---
Name: bellows
Version: 0.16.1
Summary: Library implementing EZSP
Home-page: http://github.com/zigpy/bellows
Author: Russell Cloran
Author-email: rcloran@gmail.com
License: GPL-3.0
Location: /tmp/venv/lib/python3.7/site-packages
Requires: zigpy, click-log, Click, pyserial-asyncio, pure-pcapy3, voluptuous
Required-by: 
```


`bellows -v DEBUG form`
```
debug: Using selector: EpollSelector
debug: Loading application state from /home/lex/.config/bellows/app.db
debug: Attribute id: 4 value: Jasco Products
debug: Attribute id: 5 value: 45857
debug: Checking quirks for None None (00:0d:6f:00:0b:5d:e3:b6)
debug: Considering <class 'bellows.zigbee.application.EZSPCoordinator'>
debug: Found custom device replacement for 00:0d:6f:00:0b:5d:e3:b6: <class 'bellows.zigbee.application.EZSPCoordinator'>
debug: Checking quirks for Jasco Products 45857 (00:22:a3:00:00:1b:94:bb)
debug: Considering <class 'bellows.zigbee.application.EZSPCoordinator'>
debug: Fail because endpoint list mismatch: {1} {1, 2}
debug: Attribute id: 4 value: Jasco Products
debug: Attribute id: 5 value: 45857
debug: Attribute id: 770 value: 10000
debug: Attribute id: 769 value: 1
debug: Attribute id: 768 value: 0
debug: Attribute id: 772 value: 11
debug: Attribute id: 0 value: 0
debug: Attribute id: 7 value: 1
debug: Attribute id: 0 value: 0
debug: Attribute id: 1024 value: 0
debug: Attribute id: 7 value: 1
debug: Adding group: 546, Group 0x0222
debug: Adding group: 16387, No name group 0x4003
debug: Using selector: EpollSelector
debug: Resetting EZSP
debug: Resetting ASH
debug: Sending: b'1ac038bc7e'
debug: RSTACK Version: 2 Reason: RESET_SOFTWARE frame: b'c1020b0a527e'
debug: Send command version: (4,)
debug: Sending: b'004221a850ed2c7e'
debug: Data frame: b'0142a1a8522815d2e7ae7e'
debug: Sending: b'8160597e'
debug: Application frame 0 (version) received: b'06020060'
debug: Send command version: (6,)
debug: Sending: b'7d31432157542a7d3315a67e'
debug: Data frame: b'1243a157542a13b059f4b6467e'
debug: Sending: b'82503a7e'
debug: Application frame 0 (version) received: b'06020060'
debug: Switched to EZSP protocol version 6
debug: EZSP Stack Type: 2, Stack Version: 24576
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_ADDRESS_TABLE_SIZE: 5>, 16)
debug: Sending: b'22402157547910a2593fee7e'
debug: Data frame: b'2340a157547915fbd27e'
debug: Sending: b'83401b7e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_PAN_ID_CONFLICT_REPORT_THRESHOLD: 34>, 2)
debug: Sending: b'33412157547937b059a6a77e'
debug: Data frame: b'3441a157547915904d7e'
debug: Sending: b'8430fc7e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_SECURITY_LEVEL: 13>, 5)
debug: Sending: b'4446215754797d38b75948af7e'
debug: Data frame: b'4546a157547915640c7e'
debug: Sending: b'8520dd7e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_APPLICATION_ZDO_FLAGS: 42>, 3)
debug: Sending: b'5547215754793fb1591e517e'
debug: Data frame: b'5647a157547915ce557e'
debug: Sending: b'8610be7e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_TRUST_CENTER_ADDRESS_CACHE_SIZE: 25>, 2)
debug: Sending: b'6644215754790cb059a0a67e'
debug: Data frame: b'6744a157547915e1597e'
debug: Sending: b'87009f7e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_SUPPORTED_NETWORKS: 45>, 1)
debug: Sending: b'77452157547938b3597d339e7e'
debug: Data frame: b'7045a1575479158ac67e'
debug: Sending: b'8070787e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_MAX_END_DEVICE_CHILDREN: 17>, 32)
debug: Sending: b'004a21575479049259553f7e'
debug: Data frame: b'014aa15754791573c57e'
debug: Sending: b'8160597e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_INDIRECT_TRANSMISSION_TIMEOUT: 18>, 7680)
debug: Sending: b'7d314b2157547907b24706787e'
debug: Data frame: b'124ba157547915d99c7e'
debug: Sending: b'82503a7e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_SOURCE_ROUTE_TABLE_SIZE: 26>, 16)
debug: Sending: b'2248215754790fa2594e667e'
debug: Data frame: b'2348a157547915f6907e'
debug: Sending: b'83401b7e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_STACK_PROFILE: 12>, 2)
debug: Sending: b'33492157547919b05925ba7e'
debug: Data frame: b'3449a1575479159d0f7e'
debug: Sending: b'8430fc7e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_KEY_TABLE_SIZE: 30>, 4)
debug: Sending: b'444e215754790bb6597f777e'
debug: Data frame: b'454ea157547915694e7e'
debug: Sending: b'8520dd7e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_END_DEVICE_POLL_TIMEOUT: 19>, 60)
debug: Sending: b'554f21575479068e594e147e'
debug: Data frame: b'564fa157547915c3177e'
debug: Sending: b'8610be7e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_END_DEVICE_POLL_TIMEOUT_SHIFT: 27>, 8)
debug: Sending: b'664c215754790eba593fd77e'
debug: Data frame: b'674ca157547915ec1b7e'
debug: Sending: b'87009f7e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_MULTICAST_TABLE_SIZE: 6>, 16)
debug: Sending: b'774d215754797d33a2594b317e'
debug: Data frame: b'704da15754791587847e'
debug: Sending: b'8070787e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command setConfigurationValue: (<EzspConfigId.CONFIG_PACKET_BUFFER_COUNT: 1>, 255)
debug: Sending: b'005221575479144d59302b7e'
debug: Data frame: b'0152a15754791564037e'
debug: Sending: b'8160597e'
debug: Application frame 83 (setConfigurationValue) received: b'00'
debug: Send command getConfigurationValue: (<EzspConfigId.CONFIG_APS_UNICAST_MESSAGE_COUNT: 3>,)
debug: Sending: b'7d31532157547816375a7e'
debug: Data frame: b'1253a157547815b859e4287e'
debug: Sending: b'82503a7e'
debug: Application frame 82 (getConfigurationValue) received: b'000a00'
debug: APS_UNICAST_MESSAGE_COUNT is set to 10
debug: Send command addEndpoint: (1, 260, 48879, 0, 0, 1, [], [1280])
debug: Sending: b'22502157542814b6587bf425aa54924cfe817e'
debug: Data frame: b'2350a157542815dcd87e'
debug: Sending: b'83401b7e'
debug: Application frame 2 (addEndpoint) received: b'00'
debug: Ezsp adding endpoint: [<EzspStatus.SUCCESS: 0>]
debug: Send command setConcentrator: (False, <EmberConcentratorType.HIGH_RAM_CONCENTRATOR: 65529>, 600, 1800, 2, 5, 0)
debug: Sending: b'33512157543a154ba6cc482dad57974966217e'
debug: Data frame: b'3451a157543a15d2567e'
debug: Sending: b'8430fc7e'
debug: Application frame 16 (setConcentrator) received: b'00'
debug: Set concentrator type: [<EmberStatus.SUCCESS: 0>]
debug: Send command networkInit: ()
debug: Sending: b'44562157543d05377e'
debug: Data frame: b'4556a157543d860c5a7e'
debug: Sending: b'8520dd7e'
debug: Application frame 23 (networkInit) received: b'93'
debug: Send command setInitialSecurityState: (<EmberInitialSecurityState bitmask=2820 preconfiguredKey=[90, 105, 103, 66, 101, 101, 65, 108, 108, 105, 97, 110, 99, 101, 48, 57] networkKey=[163, 2, 222, 179, 225, 144, 226, 210, 5, 31, 21, 237, 249, 92, 83, 187] networkKeySequenceNumber=0 preconfiguredTrustCenterEui64=00:00:00:00:00:00:00:00>,)
debug: Sending: b'5557215754427d31b903fd2d67cf30d325f02746c58eab57b25ec4bd3a1deedd75eed2cb8276a39460d5d2698c4623a9ec76fcfd7e'
debug: Data frame: b'5657a1575442150dbe7e'
debug: Sending: b'8610be7e'
debug: Application frame 104 (setInitialSecurityState) received: b'00'
debug: Send command formNetwork: (<EmberNetworkParameters extendedPanId=00:00:00:00:00:00:00:00 panId=0x1a96 radioTxPower=8 radioChannel=15 joinMethod=EmberJoinMethod.USE_MAC_ASSOCIATION nwkManagerId=0x0000 nwkUpdateId=0 channels=Channels.CHANNEL_25|CHANNEL_20|CHANNEL_15>,)
debug: Sending: b'66542157543415b259944a25aa550453944127abedce670bedc4720e7e'
debug: Data frame: b'6754a557543415094b7e'
debug: Sending: b'87009f7e'
debug: Data frame: b'7754b157543385a9427e'
debug: Sending: b'8070787e'
debug: Application frame 30 (formNetwork) received: b'00'
debug: Application frame 25 (stackStatusHandler) received: b'90'
debug: Send command setValue: (<EzspValueId.VALUE_STACK_TOKEN_WRITING: 7>, 1)
debug: Sending: b'70552157548112b359c4087e'
debug: Data frame: b'0055a1575481158e4b7e'
debug: Sending: b'8160597e'
debug: Application frame 171 (setValue) received: b'00'
debug: Send command getNetworkParameters: ()
debug: Sending: b'015a2157540260b17e'
debug: Data frame: b'115aa157540215b3ba836bc6d5984a7a0a542fa4edce678bfd3e9c8e58467e'
debug: Sending: b'82503a7e'
debug: Application frame 40 (getNetworkParameters) received: b'0001e31721e37fcdd833961a080f0000000000f8ff07'
debug: Send command setPolicy: (<EzspPolicyId.TC_KEY_REQUEST_POLICY: 5>, <EzspDecisionId.GENERATE_NEW_TC_LINK_KEY: 82>)
debug: Sending: b'125b2157547f10e095747e'
debug: Data frame: b'225ba157547f1530337e'
debug: Sending: b'83401b7e'
debug: Application frame 85 (setPolicy) received: b'00'
debug: Send command setPolicy: (<EzspPolicyId.APP_KEY_REQUEST_POLICY: 6>, <EzspDecisionId.ALLOW_APP_KEY_REQUESTS: 97>)
debug: Sending: b'23582157547f7d33d31f9a7e'
debug: Data frame: b'3358a157547f1571c97e'
debug: Sending: b'8430fc7e'
debug: Application frame 85 (setPolicy) received: b'00'
debug: Send command setPolicy: (<EzspPolicyId.TRUST_CENTER_POLICY: 0>, <EzspDecisionId.ALLOW_PRECONFIGURED_KEY_JOINS: 1>)
debug: Sending: b'34592157547f15b39b577e'
debug: Data frame: b'4459a157547f15a94c7e'
debug: Sending: b'8520dd7e'
debug: Application frame 85 (setPolicy) received: b'00'
debug: Send command getNodeId: ()
debug: Sending: b'455e2157540d74e97e'
debug: Data frame: b'555ea157540d15b23a117e'
debug: Sending: b'8610be7e'
debug: Application frame 39 (getNodeId) received: b'0000'
debug: Send command getEui64: ()
debug: Sending: b'565f2157540c7d3afd7e'
debug: Data frame: b'665fa157540cf3acbc9b4a4aa755e2877e'
debug: Sending: b'87009f7e'
debug: Application frame 38 (getEui64) received: b'e61ee50f006f0d00'
Device 0x0000 (00:0d:6f:00:0f:e5:1e:e6) joined the network
debug: EZSP nwk=0x0000, IEEE=00:0d:6f:00:0f:e5:1e:e6
debug: Starting EZSP watchdog
[0x0000] Requesting 'Node Descriptor'
debug: Tries remaining: 2
debug: [0x0000] Extending timeout for 0x01 request
debug: Extending timeout for 00:0d:6f:00:0f:e5:1e:e6/0x0000
debug: Send command setExtendedTimeout: (00:0d:6f:00:0f:e5:1e:e6, True)
debug: Sending: b'675c21575454f3acbc9b4a4aa75593d38e7e'
debug: Data frame: b'775ca157545482427e'
debug: Sending: b'8070787e'
debug: Application frame 126 (setExtendedTimeout) received: b''
debug: Send command sendUnicast: (<EmberOutgoingMessageType.OUTGOING_DIRECT: 0>, 0x0000, <EmberApsFrame profileId=0 clusterId=2 sourceEndpoint=0 destinationEndpoint=0 options=320 groupId=0 sequence=1>, 2, b'\x01\x00\x00')
debug: Sending: b'705d2157541e15b259944a27aa5592099d4e27aaefcd668bfd19017e'
debug: Data frame: b'005da157541e1512ce907e'
debug: Sending: b'8160597e'
debug: Application frame 52 (sendUnicast) received: b'00a0'
debug: Data frame: b'105db157546f15b259964a25aa1593499ceed8abedce9874fec763894b587e'
debug: Sending: b'82503a7e'
debug: Data frame: b'205db157546f15b25996ca25aa1593499cefd8abedce9874ecc76389fc7e7f2826668cef8fbeeb5bd5d2f6567e'
debug: Sending: b'83401b7e'
debug: Data frame: b'305db157541515b259944a27aa5592099d4e270befce677d4f7e'
debug: Sending: b'8430fc7e'
debug: Application frame 69 (incomingMessageHandler) received: b'0000000200000040010000a0ff000000ffff03010000'
debug: Received incomingMessageHandler frame with [<EmberIncomingMessageType.INCOMING_UNICAST: 0>, <EmberApsFrame profileId=0 clusterId=2 sourceEndpoint=0 destinationEndpoint=0 options=320 groupId=0 sequence=160>, 255, 0, 0x0000, 255, 255, b'\x01\x00\x00']
debug: [0x0000:zdo] ZDO request ZDOCmd.Node_Desc_req: [0x0000]
debug: [0x0000:zdo] Unsupported ZDO request:ZDOCmd.Node_Desc_req
debug: Application frame 69 (incomingMessageHandler) received: b'0000000280000040010000a1ff000000ffff110100000000408fcdab528000412c800000'
debug: Received incomingMessageHandler frame with [<EmberIncomingMessageType.INCOMING_UNICAST: 0>, <EmberApsFrame profileId=0 clusterId=32770 sourceEndpoint=0 destinationEndpoint=0 options=320 groupId=0 sequence=161>, 255, 0, 0x0000, 255, 255, b'\x01\x00\x00\x00\x00@\x8f\xcd\xabR\x80\x00A,\x80\x00\x00']
debug: [0x0000:zdo] ZDO request ZDOCmd.Node_Desc_rsp: [<Status.SUCCESS: 0>, 0x0000, <Optional byte1=0 byte2=64 mac_capability_flags=143 manufacturer_code=43981 maximum_buffer_size=82 maximum_incoming_transfer_size=128 server_mask=11329 maximum_outgoing_transfer_size=128 descriptor_capability_field=0>]
debug: [0x0000:zdo] Unsupported ZDO request:ZDOCmd.Node_Desc_rsp
debug: Application frame 63 (messageSentHandler) received: b'00000000000200000040010000a0020000'
debug: Received messageSentHandler frame with [<EmberOutgoingMessageType.OUTGOING_DIRECT: 0>, 0, <EmberApsFrame profileId=0 clusterId=2 sourceEndpoint=0 destinationEndpoint=0 options=320 groupId=0 sequence=160>, 2, <EmberStatus.SUCCESS: 0>, b'']
Shutting down ControllerApplication
debug: Cancelling watchdog
debug: Closed serial connection
```

**Comment by rossnelson** [comment]
Great thanks for the quick reply. I’ll check this out on my end.
